website/blog/2020-06-16-alluxio-datacaching.md

```diff
@@ -0,0 +1,243 @@
+This is required for multiple Facebook use-cases to improve query latency for queries that scan data from remote sources, such as HDFS. 
+We have observed significant improvements in query latencies and IO scans in our experiments. 
+<br><br>
+We found Alluxio data caching to be useful for multiple use-cases in the Facebook environment. 
+For one of the Facebook internal use cases, we observed query latencies improved by **33%** (P50), **54%** (P75), and **48%** (P95). 
+We also recorded **57%** improvement in IO for remote data source scans.
+
+# Presto Architecture
+
+Presto architecture is designed to allow data and computation to scale independently. Presto workers are responsible for executing query plan fragments 
+on the data scanned from the independent and typically remote data sources. For remote data sources, Presto workers do not store any data locally which 
+enables the computation to grow elastically.
+<br><br>
+The architecture diagram below highlights the data read paths from remote HDFS source. 
+Each worker independently reads data from the remote data source. In this blog, we will be only talking about optimizations done in the read 
+operations from the remote data source.
+<br><br>
+
+
+![](/img/blog/2020-06-16-alluxio-datacaching/presto_worker.jpg)
+
+<br>
+
+This architecture allows storage and computation to scale independently. However, scanning data from remote storage can be a costly operation, 
+and it makes achieving interactive query latency a challenge.
+
+
+# Presto + Data Caching Architecture
+
+To solve sub-second latency use cases, we decided to implement various optimizations.  One important optimization was to implement a data cache. 
+Data caching has been a traditional optimization technique to bring the working dataset closer to the compute nodes and reduce trips to remote 
+storage to save latencies and IO.
+
+The challenge was to make the data caching effective when petabytes of data get scanned from the remote data sources with no fixed pattern. 
+Another requirement for data caching to be effective was to achieve data affinity in a distributed environment like Presto.
+<br><br>
+With the addition of data caching, the Presto architecture looks like the following:
+<br><br>
+
+![](/img/blog/2020-06-16-alluxio-datacaching/presto_worker_datacache.jpg)
+<br>
+
+More on this is covered in later sections.
+
+## Soft Affinity scheduling
+
+Presto current worker node scheduling is random, it means coordinator distributes the workload randomly among workers. 
+It takes the worker load into account while distributing the load. This scheduling keeps the workload distribution uniform among workers.
+<br><br>
+However, random node scheduling does not provide any node affinity, which is required for any meaningful data caching effectiveness. 
+It is critical for the coordinator to leverage the same worker for a split which may contain the data for it in the cache.
```

Comment:
Presto’s current scheduler takes the worker load into account when distributing the splits, such scheduling strategy keeps the workload distribution uniform among workers but not necessarily guarantees any affinity, which is required for any meaningful data caching effectiveness. It is critical for the coordinator to leverage the same worker for a split when this worker may contain the data for it in the cache.